lib/sql.js

```diff
@@ -704,7 +704,15 @@ SQLConnector.prototype.save = function(model, data, options, cb) {
 
   let updateStmt = new ParameterizedSQL('UPDATE ' + this.tableEscaped(model));
   updateStmt.merge(this.buildFieldsForUpdate(model, data));
-  const whereStmt = this.buildWhere(model, where);
+  let whereStmt;
+  try {
+    whereStmt = this.buildWhere(model, where);
+  } catch (err) {
+    if (cb) {
+      cb(err);
+    }
```

Comment:
Nitpick: let's shorten this block as follows:

```js
if (cb) cb(err);
return;
```

Same comment applies to other similar places too.